### PR TITLE
checkForDangerousProps: only return true if both props have a bad value

### LINKED ISSUE #147: detect non rooted device as rooted device
I have used this lib in one of my app, but I feel like that, this lib detect some of the non rooted devices as rooted device.
I have received some review on play store saying that My device isn't rooted still then this app will not let me in.

I have received this review from Redmi Note 8 Pro

please check the issue what's wrong or let me know possible reason behind this behaviour.

### LINKED ISSUE #181: False positive - Dangerous props
Hi,

I have a fairphone 3+ with a custom rom (lineageOS microG), but I don't have, and have never, had root. Your test app triggers on 'dangerous props', but I'm not really sure what that check does, so even though it looks like a false positive, it might be due to my custom rom!

Let me know if I can test anything for you guys

### PR DESCRIPTION
**Pull request by Tachi107**
See #147 for details :)
Fixes #147, fixes #181

### CODE REVIEW

**Comment by Tachi107** [comment]
@stealthcopter sorry for the ping, but I see you've been involved in #147, so you might be interested in reviewing this. Thanks!

**Review by eksencncozay — COMMENTED** [review]
Oz

**Comment by scottyab** [comment]
Hey @Tachi107 Sorry it's taken so very long to respond. I have a window between jobs at the moment and attempting to address these PRs, update the documentation around the root checks and release a new version of rootbeer. In terms of this PR the checkForDangerousProps check is working as intended it'll flag for any of the properties not matching the expected value so changes in this PR are not aligned here so closing this PR. But thank you for your efforts here.  

There's a different question in terms of whether the `ro.debuggable` is truly an indication of root. In one sense it's not, on the other it's an indication the device is not in a production-ready state so the plan is to keep the functionality as it for the time being.   

To clarify @eksencncozay is not a maintainer of this project. Perhaps they can add some context to their review/comment/ 

Again sorry it's taken so long to respond.

**Comment by Tachi107** [comment]
Hi Scott, thanks for the reply! Don't worry about the time it took to respond, good things take time :)

I do agree that, looking back at this patch, only returning true in both properties are in an unexpected state is wrong. I am now of the opinion that the function should return true only if `ro.secure` is set to `0`.

The purpose of `checkForDangerousProps` is to check for properties which might imply root access. As thoroughly discussed in #147, `ro.secure` likely implies root, while `ro.debuggable` is unrelated to this.

Yes, `ro.debuggable` might imply that the phone is in a non-production-ready state, but there's already an official way to check for "certified devices": SafetyNet, or more recently, Google Play Integrity API.

And as seen in #147, checking for `ro.debuggable` for the reason of checking non-production-readyness is flawed, as many devices are sold with the property enabled.

In short: a root checking library should only check for `ro.secure`, an integrity checking library cannot rely on any of them (and you can't beat Google's Play Integrity API in this)

Hope this makes sense to you! Thanks again and bye :)

**Comment by scottyab** [comment]
>In short: a root checking library should only check for ro.secure, an integrity checking library cannot rely on any of them (and you can't beat Google's Play Integrity API in this)

Thanks for the extra comments. Yes, agree with this. Given removing the ro.debuggable maybe something users are relying on I think it would be better placed to add this to new major version of the library.
